store/multistore/store.go

```diff
@@ -0,0 +1,301 @@
+// CommitInfo. An error is returned if the latest CommitInfo or version cannot
+// be retrieved.
+func (s *Store) GetLatestVersion() (uint64, error) {
+	lastCommitID, err := s.LastCommitID()
+	if err != nil {
+		return 0, err
+	}
+
+	return uint64(lastCommitID.Version), nil
+}
+
+func (s *Store) GetProof(storeKey string, version uint64, key []byte) (*ics23.CommitmentProof, error) {
+	sc, ok := s.scStores[storeKey]
+	if !ok {
+		return nil, fmt.Errorf("SC store with key %s not mounted", storeKey)
+	}
+
+	return sc.GetProof(version, key)
+}
+
+func (s *Store) GetSCStore(storeKey string) *commitment.Database {
+	panic("not implemented!")
+}
+
+func (s *Store) LoadLatestVersion() error {
+	lv, err := s.GetLatestVersion()
+	if err != nil {
+		return err
+	}
+
+	return s.loadVersion(lv, nil)
+}
+
+func (s *Store) LoadVersion(v uint64) (err error) {
+	return s.loadVersion(v, nil)
+}
+
+func (s *Store) loadVersion(v uint64, upgrades *v1types.StoreUpgrades) (err error) {
+	s.logger.Debug("loading version", "version", v)
+
+	for sk, sc := range s.scStores {
+		if loadErr := sc.LoadVersion(v); loadErr != nil {
+			err = errors.Join(err, fmt.Errorf("failed to load version %d for %s: %w", v, sk, loadErr))
+		}
+	}
+
+	// TODO: Complete this method to handle upgrades. See legacy RMS loadVersion()
+	// for reference.
+
+	return err
+}
+
+func (s *Store) WorkingHash() []byte {
+	storeInfos := make([]v1types.StoreInfo, 0, len(s.scStores))
+
+	for sk, sc := range s.scStores {
+		if _, ok := s.removalMap[sk]; ok {
+			storeInfos = append(storeInfos, v1types.StoreInfo{
+				Name: sk,
+				CommitId: v1types.CommitID{
+					Hash: sc.WorkingHash(),
+				},
+			})
+		}
+	}
+
+	sort.SliceStable(storeInfos, func(i, j int) bool {
+		return storeInfos[i].Name < storeInfos[j].Name
+	})
+
+	return v1types.CommitInfo{StoreInfos: storeInfos}.Hash()
+}
+
+func (s *Store) SetCommitHeader(h CommitHeader) {
+	s.commitHeader = h
+}
+
+func (s *Store) Commit() ([]byte, error) {
+	var previousHeight, version uint64
+	if s.lastCommitInfo.GetVersion() == 0 && s.initialVersion > 1 {
+		// This case means that no commit has been made in the store, we
+		// start from initialVersion.
+		version = s.initialVersion
+	} else {
+		// This case can means two things:
+		//
+		// 1. There was already a previous commit in the store, in which case we
+		// 		increment the version from there.
+		// 2. There was no previous commit, and initial version was not set, in which
+		// 		case we start at version 1.
+		previousHeight = uint64(s.lastCommitInfo.GetVersion())
+		version = previousHeight + 1
+	}
+
+	if s.commitHeader.GetHeight() != version {
+		s.logger.Debug("commit header and version mismatch", "header_height", s.commitHeader.GetHeight(), "version", version)
+	}
+
+	// remove and close all SC stores marked for removal
+	for sk := range s.removalMap {
+		if sc, ok := s.scStores[sk]; ok {
+			if err := sc.Close(); err != nil {
+				return nil, err
+			}
```

Comment:
This to me looks like it could cause inconsistency, why don't we just fully delete all the marked SC stores and use errors.Join(err, e) then after this loop, return the error?
```go
for sk := range s.removalMap {
     sc, ok := s.scStores[sk]
     if !ok {
         continue
     }

     if ce := sc.Close(); ce != nil {
         err = errors.Join(err, ce)
     }
     delete(s.scStores, sk)
}

if err != nil {
    return nil, err
}
```